api/v1alpha4/azuremachine_webhook.go

```diff
@@ -81,44 +83,88 @@ func (m *AzureMachine) ValidateUpdate(oldRaw runtime.Object) error {
 	var allErrs field.ErrorList
 	old := oldRaw.(*AzureMachine)
 
-	if errs := ValidateImage(m.Spec.Image, field.NewPath("image")); len(errs) > 0 {
-		allErrs = append(allErrs, errs...)
+	if !reflect.DeepEqual(m.Spec.Image, old.Spec.Image) {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "Image"),
+				m.Spec.Image, "field is immutable"),
+		)
 	}
 
-	if errs := ValidateOSDisk(m.Spec.OSDisk, field.NewPath("osDisk")); len(errs) > 0 {
-		allErrs = append(allErrs, errs...)
+	if !reflect.DeepEqual(m.Spec.Identity, old.Spec.Identity) {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "Identity"),
+				m.Spec.Identity, "field is immutable"),
+		)
 	}
 
-	if errs := ValidateSSHKey(m.Spec.SSHPublicKey, field.NewPath("sshPublicKey")); len(errs) > 0 {
-		allErrs = append(allErrs, errs...)
+	if !reflect.DeepEqual(m.Spec.UserAssignedIdentities, old.Spec.UserAssignedIdentities) {
```

Comment:
This will fail when the items in the array are re arranged. Are we okay with that?

Reply:
Same for `DataDisks`.

Reply:
Ideally, we want these fields (`DataDisks` and `UserAssignedIdentities`) should not be updated, which means values used while creation and updation should be same (may be we can copy from creation yaml)
But anyways please suggest whether we should allow rearranging slice items?

Reply:
Yeah, I think it's a fair assumption that user should not re-arrange the items.